File: Assets/Editor/SceneAPI/Modules/SceneManagementModule.cs
using System;
using System.IO;
using System.Linq;
using System.Net;
using Newtonsoft.Json;
using UnityEditor;
using UnityEditor.SceneManagement;

namespace SceneAPI.Modules
{
    public static class SceneManagementModule
    {
        public static string OpenScene(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);
                
                string scenePath = data?.scenePath;
                
                if (string.IsNullOrEmpty(scenePath))
                {
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = false, 
                        error = "Scene path is required" 
                    });
                }

                if (EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo())
                {
                    EditorSceneManager.OpenScene(scenePath);
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = true, 
                        message = $"Scene opened: {scenePath}" 
                    });
                }

                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = "Failed to save current scene or user cancelled" 
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = $"Error opening scene: {ex.Message}" 
                });
            }
        }

        public static string GetBuildScenes()
        {
            try
            {
                var buildScenes = EditorBuildSettings.scenes
                    .Select(scene => new
                    {
                        path = scene.path,
                        enabled = scene.enabled,
                        guid = scene.guid.ToString()
                    })
                    .ToArray();

                return JsonConvert.SerializeObject(new 
                { 
                    scenes = buildScenes,
                    totalCount = buildScenes.Length
                }, Formatting.Indented);
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new 
                { 
                    error = $"Error getting build scenes: {ex.Message}" 
                });
            }
        }

        public static string AddSceneToBuild(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);
                
                string scenePath = data?.scenePath;
                
                if (string.IsNullOrEmpty(scenePath))
                {
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = false, 
                        error = "Scene path is required" 
                    });
                }

                var scenes = EditorBuildSettings.scenes.ToList();
                if (!scenes.Any(s => s.path == scenePath))
                {
                    scenes.Add(new EditorBuildSettingsScene(scenePath, true));
                    EditorBuildSettings.scenes = scenes.ToArray();
                    
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = true, 
                        message = $"Scene added to build: {scenePath}" 
                    });
                }

                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = "Scene already in build settings" 
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = $"Error adding scene to build: {ex.Message}" 
                });
            }
        }

        public static string RemoveSceneFromBuild(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);
                
                string scenePath = data?.scenePath;
                
                if (string.IsNullOrEmpty(scenePath))
                {
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = false, 
                        error = "Scene path is required" 
                    });
                }

                var scenes = EditorBuildSettings.scenes.ToList();
                var sceneToRemove = scenes.FirstOrDefault(s => s.path == scenePath);
                
                if (sceneToRemove != null)
                {
                    scenes.Remove(sceneToRemove);
                    EditorBuildSettings.scenes = scenes.ToArray();
                    
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = true, 
                        message = $"Scene removed from build: {scenePath}" 
                    });
                }

                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = "Scene not found in build settings" 
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = $"Error removing scene from build: {ex.Message}" 
                });
            }
        }

        private static string GetRequestBody(HttpListenerContext context)
        {
            using (var reader = new StreamReader(context.Request.InputStream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}

File: Assets/PythonMCP/modules/scene_management_module.py
import requests
import json
from typing import Dict, Optional

class SceneManagementModule:
    def __init__(self, base_url: str):
        self.base_url = base_url
    
    def open_scene(self, scene_path: str) -> Dict:
        """Открывает указанную сцену"""
        try:
            if not scene_path:
                return {
                    "success": False,
                    "action": "open_scene",
                    "error": "scene_path is required"
                }
            
            response = requests.post(
                f"{self.base_url}/scene/open", 
                json={"scenePath": scene_path}
            )
            response.raise_for_status()
            result = response.json()
            
            return {
                "success": result.get("success", False),
                "action": "open_scene",
                "data": result if result.get("success") else None,
                "error": result.get("error")
            }
            
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "open_scene",
                "error": f"Request error: {str(e)}"
            }
    
    def get_build_scenes(self) -> Dict:
        """Получает список сцен в настройках сборки"""
        try:
            response = requests.get(f"{self.base_url}/build/scenes")
            response.raise_for_status()
            result = response.json()
            
            return {
                "success": True,
                "action": "get_build_scenes",
                "data": result,
                "error": result.get("error") if "error" in result else None
            }
            
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "get_build_scenes",
                "error": f"Request error: {str(e)}"
            }
    
    def add_scene_to_build(self, scene_path: str) -> Dict:
        """Добавляет сцену в настройки сборки"""
        try:
            if not scene_path:
                return {
                    "success": False,
                    "action": "add_scene_to_build",
                    "error": "scene_path is required"
                }
            
            response = requests.post(
                f"{self.base_url}/build/scenes/add", 
                json={"scenePath": scene_path}
            )
            response.raise_for_status()
            result = response.json()
            
            return {
                "success": result.get("success", False),
                "action": "add_scene_to_build",
                "data": result if result.get("success") else None,
                "error": result.get("error")
            }
            
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "add_scene_to_build",
                "error": f"Request error: {str(e)}"
            }
    
    def remove_scene_from_build(self, scene_path: str) -> Dict:
        """Удаляет сцену из настроек сборки"""
        try:
            if not scene_path:
                return {
                    "success": False,
                    "action": "remove_scene_from_build",
                    "error": "scene_path is required"
                }
            
            response = requests.delete(
                f"{self.base_url}/build/scenes/remove", 
                json={"scenePath": scene_path}
            )
            response.raise_for_status()
            result = response.json()
            
            return {
                "success": result.get("success", False),
                "action": "remove_scene_from_build",
                "data": result if result.get("success") else None,
                "error": result.get("error")
            }
            
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "remove_scene_from_build",
                "error": f"Request error: {str(e)}"
            }